frontend/src/components/recipepage.jsx

```diff
@@ -0,0 +1,100 @@
+import './recipepage.css';
+import React, {useState, useEffect} from "react";
+
+  
+export default function Recipepage({name,  desc, image, ingredients,instructions}){
+  const [newIngredient, setNewIngredient] = React.useState('');
+  const [newInstruction, setNewInstruction] = React.useState('');
+  
+
+const addIngredient = () => {
+    var ul = document.getElementById("ingredientList");
+    var li = document.createElement("li");
+    li.appendChild(document.createTextNode(newIngredient));
+    ul.appendChild(li);
+  }
+  
+  const addInstruction = () => {
+    var ol = document.getElementById("instructionList");
+    var li = document.createElement("li");
+    li.appendChild(document.createTextNode(newInstruction));
+    ol.appendChild(li);
+  }
+
+  useEffect(() => {
```

Comment:
You don't need to have a useEffect for POST or PUT requests. Remember, By using the useEffect Hook, you tell React that your component needs to do something after a page renders (or a variable changes).

However, for POST and PUT requests, they should be ran whenever the addIngredient or addInstruction button is clicked, so we can just call a function which includes our fetch request whenever that button is clicked